### ISSUE TITLE
1.8.7 can't connect

### ISSUE THREAD

**Issue by j2l** [supabase/cli#539]
# Bug report
## Describe the bug
On ubuntu 22.04, I'm running supabase docker with port 5432 open for supabase-db on docker_default.
postgres container **is accessible by dbeaver** using same credentials and port.
Also `npx openapi-typescript http://localhost:8000/rest/v1/?apikey=<key> --output src/componets/database.types.ts` works (AFAIU result is dfferent from supabase-cli).
But not direct access to postgres container using this tool, I'm getting:
```
2022/10/18 20:14:30 Sent Header: Host [docker]
2022/10/18 20:14:30 Sent Header: User-Agent [Go-http-client/1.1]
2022/10/18 20:14:30 Send Done
2022/10/18 20:14:30 Recv First Byte
2022/10/18 20:14:30 Sent Header: Host [docker]
2022/10/18 20:14:30 Sent Header: User-Agent [Go-http-client/1.1]
2022/10/18 20:14:30 Send Done
2022/10/18 20:14:30 Recv First Byte
2022/10/18 20:14:30 Sent Header: Host [docker]
2022/10/18 20:14:30 Sent Header: User-Agent [Go-http-client/1.1]
2022/10/18 20:14:30 Sent Header: Content-Length [1796]
2022/10/18 20:14:30 Sent Header: Content-Type [application/json]
2022/10/18 20:14:30 Send Done
2022/10/18 20:14:30 Recv First Byte
2022/10/18 20:14:30 Sent Header: Host [docker]
2022/10/18 20:14:30 Sent Header: User-Agent [Go-http-client/1.1]
2022/10/18 20:14:30 Sent Header: Content-Length [0]
2022/10/18 20:14:30 Sent Header: Content-Type [text/plain]
2022/10/18 20:14:30 Send Done
2022/10/18 20:14:30 Recv First Byte
2022/10/18 20:14:30 Sent Header: Host [docker]
2022/10/18 20:14:30 Sent Header: User-Agent [Go-http-client/1.1]
2022/10/18 20:14:30 Send Done
2022/10/18 20:14:30 Recv First Byte
/usr/src/app/bin/src/server/app.js:74
            throw new Error(schemasError.message);
                  ^

Error: connect ECONNREFUSED 127.0.0.1:5432
    at /usr/src/app/bin/src/server/app.js:74:19
    at Generator.next (<anonymous>)
    at fulfilled (/usr/src/app/bin/src/server/app.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022/10/18 20:14:31 Sent Header: Host [docker]
2022/10/18 20:14:31 Sent Header: User-Agent [Go-http-client/1.1]
2022/10/18 20:14:31 Send Done
2022/10/18 20:14:31 Recv First Byte
Error: error running container
```

## To Reproduce
- run supabase containers
- `sudo supabase gen types typescript --db-url postgres://postgres:password@localhost:5432/postgres > src/components/database.types.ts --debug`

## Expected behavior
populate `database.types.ts`

## System information
- OS: ubuntu 22.04 (popos)
- Version of supabase-cli: 1.8.7

## Additional context
I tried replacing locahost with 0.0.0.0 and the internal container IP (on docker-default network), without luck.
The CLI downloaded the additional image `public.ecr.aws/t3w2s2c9/postgres-meta:v0.50.2` and creates a temporary container, but it's on `bridge` network by default. Even by adding this container to the main network it doesn't connect.

postgres is stock from docker-compose:
```
  db:
    container_name: supabase-db
    image: supabase/postgres:14.1.0.21
    command: postgres -c config_file=/etc/postgresql/postgresql.conf
    restart: unless-stopped
    ports:
      - ${POSTGRES_PORT}:5432
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    volumes:
      # - ./volumes/db/data:/var/lib/postgresql/data
      - ./volumes/db/init:/docker-entrypoint-initdb.d
```

EDIT: hmmm, my password is 45 characters long, is there a limitation?

**Comment by soedirgo** [comment]
The `postgres-meta` container (which runs the typegen) uses a temporary bridge network, so `localhost` wouldn't work as expected. Can you try using `host.docker.internal` instead of `localhost`?

**Comment by j2l** [comment]
Thanks @soedirgo 
I get `Error: getaddrinfo ENOTFOUND host.docker.internal`

**Comment by soedirgo** [comment]
Sorry, just realized you're on Linux. How about [`172.17.0.1`](https://stackoverflow.com/a/48547074/12396224)?

**Comment by j2l** [comment]
I also tried the container internal IP, it's spinning for 1min and quit, that's how I found the temporary running container, changed its network to docker_default while it was running, without luck.

EDIT:  172.17.0.1 works!

Thank you very much for taking the time @soedirgo ! I didn't know about this gateway IP.
Closing this issue.

Bonus question: Is there a way to shut done the openapi schema URL to prevent attacks using anon key?
